hashintel/hash
# Update Code block for Block Protocol 0.3

## 🌟 What is the purpose of this PR?

Updates the Code block for the forthcoming `0.3` version of the Block Protocol specification. 

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203543021352032/1203673154406411/f) _(internal)_

## 🔍 What does this change?

- Installs canary version of block packages which were published from https://github.com/blockprotocol/blockprotocol/pull/879
- Ignores these packages in the yarn dependency check to avoid having to touch other packages in this PR
- Reference an externally-hosted `schema` in `package.json` which replaces the previous local definition of the block's entity type (previously `BlockEntityProperties`) – this schema is read and an equivalent type – now `RootEntity` - produced via the new `yarn codegen` script
- Update the block's props for the new shape of `graph`, and reads `blockEntity` from the new `blockEntitySubgraph` (which replaces `blockEntity, `linkGroups` and `linkedEntities`, although the latter two are irrelevant to this block)
- Adds some extra code to deal with the fact that property keys are now URIs and are a bit more of a pain to deal with (they are invalid variable names and therefore cannot be deconstructed from objects)
- Updates the props sent to `MockBlockDock`

## 🐾 Next steps

- #1823 makes this block loadable in HASH.
- Update more blocks to `0.3`

## ❓ How to test this?



1.  `yarn install && yarn dev`
2. Update some properties in the code block, check it's reflected in the mock datastore (bottom tab)
3. Edit some properties in the mock datastore, check it's reflected in the code block
